### PR TITLE
Rapp module

### PR DESCRIPTION
**Pull request by jihoonl**
rapp manager has been upgraded.

https://github.com/robotics-in-concert/rocon_app_platform/issues/167
https://github.com/robotics-in-concert/rocon_app_platform/issues/153
https://github.com/robotics-in-concert/rocon_app_platform/issues/139
https://github.com/robotics-in-concert/rocon_app_platform/issues/137

### CODE REVIEW

**Comment by stonier** [comment]
I am making fixes on the command line, but check your source/deb workspaces. I think you have old sources there (you have imports for rocon_utilities which is long gone and creating errors).

File: rocon_app_manager/src/rocon_app_manager/rapp_manager.py [hunk 1]
```diff
@@ -280,10 +274,16 @@ def _determine_runnable_rapps(self, rapps):
 
     def _process_invite(self, req):
         '''
-          @todo This needs better arranging for logic. Currently it ignores whitelists/blacklists if the local_remote_controllers
+          :todo: This needs better arranging for logic. Currently it ignores whitelists/blacklists if the local_remote_controllers
```
Comment (stonier) [inline rocon_app_manager/src/rocon_app_manager/rapp_manager.py:270]: @jihoonl does this work like this? I had to do:

```
.. todo:: tighten up the name pattern matching
```

in rocon_uri to get something out in sphinx.

Comment (jihoonl) [inline rocon_app_manager/src/rocon_app_manager/rapp_manager.py:270]: It works similar way. 
Search for todo
http://192.168.10.129/rocondoc_dev/rocon_app_manager/html/rapp_manager_api.html

**Comment by jihoonl** [comment]
These issues are resolved with updates. Syncing connection key between gateway and app manager 

https://github.com/robotics-in-concert/rocon_app_platform/issues/172
https://github.com/robotics-in-concert/rocon_app_platform/issues/171
https://github.com/robotics-in-concert/rocon_app_platform/issues/169

**Comment by jihoonl** [comment]
https://github.com/robotics-in-concert/rocon_app_platform/issues/179

**Comment by bit-pirate** [comment]
@jihoonl is this ready to be merged or are you still working on this?

**Comment by jihoonl** [comment]
Nope. Done unless there is bug

**Pull request merged**